modules/secure_storage_connector/main.tf

```diff
@@ -0,0 +1,55 @@
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_client_config" "current" {
+}
+
+data "azurerm_resource_group" "group" {
+  name = var.resource_group_name
+}
+
+resource "azurerm_user_assigned_identity" "default" {
+  name                = "${var.namespace}-identity"
+  location            = data.azurerm_resource_group.group.location
+  resource_group_name = data.azurerm_resource_group.group.name
+}
+
+resource "azurerm_federated_identity_credential" "app" {
+  parent_id           = azurerm_user_assigned_identity.default.id
+  name                = "${var.namespace}-federated-credential"
+  resource_group_name = data.azurerm_resource_group.group.name
+  audience            = ["api://AzureADTokenExchange"]
+  issuer              = var.oidc_issuer_url
+  subject             = "system:serviceaccount:default:wandb-app"
+}
+
+module "storage" {
+  depends_on = [azurerm_user_assigned_identity.default]
+
+  source = "../storage"
```

Comment:
I don't have time to read the code but it feels wrong to laterally import a terraform module

Reply:
gotcha, yea I originally just repeated all of the storage module code here, but saw we were doing this import in the  byob module, so I started doing here